tests/core/pyspec/eth2spec/test/altair/sync_protocol/test_light_client_sync.py

```diff
@@ -0,0 +1,424 @@
+from typing import (Any, Dict, List)
+
+from eth_utils import encode_hex
+from eth2spec.test.context import (
+    spec_state_test_with_matching_config,
+    with_presets,
+    with_altair_and_later,
+)
+from eth2spec.test.helpers.attestations import (
+    next_slots_with_attestations,
+    state_transition_with_full_block,
+)
+from eth2spec.test.helpers.constants import MINIMAL
+from eth2spec.test.helpers.light_client import (
+    get_sync_aggregate,
+)
+from eth2spec.test.helpers.state import (
+    next_slots,
+    transition_to,
+)
+
+
+def setup_test(spec, state):
+    class LightClientSyncTest(object):
+        steps: List[Dict[str, Any]]
+        genesis_validators_root: spec.Root
+        store: spec.LightClientStore
+
+    test = LightClientSyncTest()
+    test.steps = []
+
+    yield "genesis_validators_root", "meta", "0x" + state.genesis_validators_root.hex()
+    test.genesis_validators_root = state.genesis_validators_root
+
+    next_slots(spec, state, spec.SLOTS_PER_EPOCH * 2 - 1)
+    trusted_block = state_transition_with_full_block(spec, state, True, True)
+    trusted_block_root = trusted_block.message.hash_tree_root()
+    bootstrap = spec.create_light_client_bootstrap(state)
+    yield "trusted_block_root", "meta", "0x" + trusted_block_root.hex()
+    yield "bootstrap", bootstrap
+    test.store = spec.initialize_light_client_store(trusted_block_root, bootstrap)
+
+    return test
+
+
+def finalize_test(test):
+    yield "steps", test.steps
+    yield "expected_finalized_header", test.store.finalized_header
+    yield "expected_optimistic_header", test.store.optimistic_header
+
+
+def get_update_file_name(spec, update):
+    if spec.is_sync_committee_update(update):
+        suffix1 = "s"
+    else:
+        suffix1 = "x"
+    if spec.is_finality_update(update):
+        suffix2 = "f"
+    else:
+        suffix2 = "x"
+    return f"update_{encode_hex(update.attested_header.hash_tree_root())}_{suffix1}{suffix2}"
+
+
+def emit_slot(test, spec, state):
+    current_slot = state.slot
+    yield from []
```

Comment:
we can just
- remove this line and
- remove `yield from` from every `emit_slot` call

Reply:
Yeah, that is possible, but allowing the consistent `yield from` notion (even if not technically necessary) make the functions that append to the test script (`setup_test`, `emit_slot`, `emit_update`, `finish_test`) similar in use, which helps with readability of the usage code.